src/Tools/dotnet-monitor/Exceptions/ExceptionsOperation.cs

```diff
@@ -85,12 +87,51 @@ public Task StopAsync(CancellationToken token)
 
         private async Task WriteJson(Stream stream, IReadOnlyList<IExceptionInstance> instances, CancellationToken token)
         {
-            foreach (IExceptionInstance instance in instances)
+            foreach (IExceptionInstance instance in FilterExceptions(_configuration, instances))
             {
                 await WriteJsonInstance(stream, instance, token);
             }
         }
 
+        internal static List<IExceptionInstance> FilterExceptions(ExceptionsConfigurationSettings configuration, IReadOnlyList<IExceptionInstance> instances)
+        {
+            List<IExceptionInstance> filteredInstances = new List<IExceptionInstance>();
+            foreach (IExceptionInstance instance in instances)
+            {
+                if (FilterException(configuration, instance))
+                {
+                    filteredInstances.Add(instance);
+                }
+            }
+
+            return filteredInstances;
+        }
+
+        internal static bool FilterException(ExceptionsConfigurationSettings configuration, IExceptionInstance instance)
+        {
+            bool shouldInclude = true;
+            if (configuration.Include.Count > 0)
+            {
+                shouldInclude = false;
+                // filter out exceptions that don't match the filter
+                if (configuration.ShouldInclude(instance))
+                {
+                    shouldInclude = true;
+                }
+            }
+
+            if (configuration.Exclude.Count > 0)
```

Comment:
If you process the exclude list first and the include list second, then you don't need a local variable to maintain whether the exception should ultimately be included.